sips/committee_consensus.md

```diff
@@ -0,0 +1,480 @@
+	senderIDStartPos = roleTypeStartPos + roleTypeSize
+)
+
+// MessageID is used to identify and route messages to the right validator and Runner
+type MessageID [56]byte
+
+func (msg MessageID) GetDomain() DomainType {
+	return msg[domainStartPos : domainStartPos+domainSize]
+}
+
+func (msg MessageID) GetRoleType() RunnerRole {
+	roleByts := msg[roleTypeStartPos : roleTypeStartPos+roleTypeSize]
+	return BeaconRole(binary.LittleEndian.Uint32(roleByts))
+}
+
+func (msg MessageID) GetSenderID() []byte {
+	return msg[senderIDStartPos : senderIDStartPos+senderIDSize]
+}
+
+func parseMessageID(messageID MessageID) (DomainType, BeaconRole, []byte) {
+	domain := messageID.GetDomain()
+	role := messageID.GetRoleType()
+	if role == BNRoleSyncCommitteeContribution {
+		return domain, BNRoleAttesterOrSyncCommittee, messageID.GetSenderID()[16:]
+	}
+	return domain, role, messageID.GetSenderID()
+}
+```
+
+
+### Consensus Data
+
+We note that the data needed for a consensus execution is the same for all validators. Thus, we can create a `BeaconVote` object that will hold the data for all validators. The data needed for the `SyncCommittee` role is a subset of the data needed for the `Attestation` role. Thus we can always query the beacon node for attestation data and pass this data to relevant runners.
+
+`ConsensusData` currently holds the `duty` field to make sure that all committee members agree on committee information. However, if there is a difference between committee members there is no guarantee that a run will be triggered on the first place. This is because different views may cause [different shuffles](https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/beacon-chain.md#compute_committee). Therefore we can rely on local view of `duty`, and keep the field empty. 
+
+
+
+### Value Check
+
+Now that consensus handles value for multiple validators there is the issue that a value may be considered to be slashable by some validators but not by others. This can happen if some operators have different views of the ethereum chain. During the consensus value-check process we check whether a value is considered slashable by at least one of the validators. If so the check fails. Since we use one value for all validators, the `CommitteeIndex` takes a value of `MaxUInt64` to indicate this ambiguous state.
+
+```go
+func BeaconVoteValueCheckF(
+	signer types.BeaconSigner,
+	slot phase0.Slot,
+	sharePublicKeys []types.ShareValidatorPK,
+	estimatedCurrentEpoch phase0.Epoch,
+) qbft.ProposedValueCheckF {
+	return func(data []byte) error {
+		bv := types.BeaconVote{}
+		if err := bv.Decode(data); err != nil {
+			return errors.Wrap(err, "failed decoding beacon vote")
+		}
+
+		if bv.Target.Epoch > estimatedCurrentEpoch+1 {
+			return errors.New("attestation data target epoch is into far future")
+		}
+
+		if bv.Source.Epoch >= bv.Target.Epoch {
+			return errors.New("attestation data source >= target")
+		}
+
+		attestationData := &phase0.AttestationData{
```

Comment:
let's separate it to a different function because we might have multiple checks (not just attestation)

Reply:
only attestation is slashable
I don't see anything to fix